pkg/preparer/setup.go

```diff
@@ -399,13 +398,22 @@ func New(preparerConfig *PreparerConfig, logger logging.Logger) (*Preparer, erro
 		finishExec = preparerConfig.PodProcessReporterConfig.FinishExec()
 	}
 
+	var hooksManifest manifest.Manifest
+	var hooksPod *pods.Pod
+	if preparerConfig.HooksManifest != "" {
+		hooksManifest, err = manifest.FromBytes([]byte(preparerConfig.HooksManifest))
+		if err != nil {
+			return nil, util.Errorf("Could not parse configured hooks manifest: %s", err)
+		}
+		hooksPodFactory := pods.NewHookFactory(filepath.Join(preparerConfig.PodRoot, "hooks"), preparerConfig.NodeName)
+		hooksPod = hooksPodFactory.NewHookPod(hooksManifest.ID())
+	}
```

Comment:
Does the lack of an else clause here preserve some "hookless" behaviour we were supporting before? I'm asking because I'd like to see an error check (& crash) here instead of proceeding.